src/meta-srv/src/procedure/alter_table.rs

```diff
@@ -75,11 +73,57 @@ impl AlterTableProcedure {
         Ok(AlterTableProcedure { context, data })
     }
 
+    // Checks whether the table exists.
+    async fn on_prepare(&mut self) -> Result<Status> {
+        let request = &self.data.alter_table_request;
+
+        let manager = &self.context.table_metadata_manager;
+
+        if let AlterKind::RenameTable { new_table_name } = &request.alter_kind {
+            let exist = manager
+                .table_name_manager()
+                .exist(TableNameKey::new(
+                    &request.catalog_name,
+                    &request.schema_name,
+                    new_table_name,
+                ))
+                .await
+                .context(TableMetadataManagerSnafu)?;
+
+            ensure!(
+                !exist,
+                error::TableAlreadyExistsSnafu {
+                    table_name: request.table_ref().to_string()
+                }
+            )
+        }
+
+        let exist = manager
+            .table_name_manager()
+            .exist(TableNameKey::new(
+                &request.catalog_name,
+                &request.schema_name,
+                &request.table_name,
+            ))
+            .await
+            .context(TableMetadataManagerSnafu)?;
+
+        ensure!(
+            exist,
+            error::TableNotFoundSnafu {
+                name: request.table_ref().to_string()
+            }
+        );
```

Comment:
We can execute a batch exists (or a txn) to reduce a call to the remote storage.